### ISSUE TITLE
Datepicker popup onFocus: cannot read property 0 of undefined

### ISSUE THREAD

**Issue by ginna-baker** [angular-ui/bootstrap#5107]
Version 0.14.3

Using datepicker-template-url in combination with [day.html] (https://github.com/angular-ui/bootstrap/blob/master/template/datepicker/day.html) from angular-ui/bootstrap templates.

**Comment by icfantv** [comment]
@ginna-baker, please do not use the issues forum for support-related requests.  Rather, please follow the instructions [here](https://github.com/angular-ui/bootstrap#need-help) as support-related questions are best served by and for the community.  The issues forum is reserved for bugs.  Thanks.

That said, please extend us the courtesy of a minimal, reproducible plunker (as documented) so we can investigate.  It's difficult, if not impossible, to diagnose problems with no information.  Thanks.

**Comment by dimbslmh** [comment]
@icfantv I experienced the same bug and made a plnkr here

**Comment by icfantv** [comment]
@dimbslmh at first glance it appears you're trying to focus an element that's not yet in the DOM.

**Comment by dimbslmh** [comment]
@icfantv Yeah, my bad. After some more testing, it seems this error occurred because I was using $provide.decorator to override the template. And I found out that when using datepicker-template-url, the DatepickerPopup does not get focus the first time. At least my console is error free now. I updated my [plnkr](http://plnkr.co/edit/UWryYIJ6gG8jgYfOwiMX?p=preview) to show the on open focus "bug" using the datepicker-template-url. Thanks
